src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -465,15 +465,24 @@ internal bool TryGetValue(int index, out DateTime value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (JsonHelpers.TryParseAsISO(segment, out DateTime tmp, out int bytesConsumed) &&
-                segment.Length == bytesConsumed)
+            if (!JsonReaderHelper.IsValidDateTimeOffsetParseLength(segment.Length))
             {
-                value = tmp;
-                return true;
+                value = default;
+                return false;
             }
 
+            // Segment needs to be unescaped
+            if (row.HasComplexChildren)
+            {
+                return JsonReaderHelper.TryGetEscapedDateTime(segment, out value);
+            }
+
+            Debug.Assert(segment.IndexOf(JsonConstants.BackSlash) == -1);
+
             value = default;
-            return false;
+            return (segment.Length <= JsonConstants.MaximumDateTimeOffsetParseLength)
+                && JsonHelpers.TryParseAsISO(segment, out value, out int bytesConsumed)
```

Comment:
Open question: Do we need to prevent the `value` out param from containing an instantiated `DateTime` instance if the `segment.Length == bytesConsumed` check fails and we're about to return false?

(This question applies to a few other places throughout this file.)

Reply:
Yes, I think we do. I'll follow up with a namespace-wide change following input from @ahsonkhan @bartonjs.

Reply:
I'd say yes: it should emit default(T) when false is returned, and should not emit a semantic value.

Reply:
Agreed.